src/module_loader.rs

```diff
@@ -97,9 +129,21 @@ impl ModuleLoader for RustyLoader {
         &self,
         specifier: &str,
         referrer: &str,
-        _kind: deno_core::ResolutionKind,
+        kind: deno_core::ResolutionKind,
     ) -> Result<ModuleSpecifier, anyhow::Error> {
-        let url = deno_core::resolve_import(specifier, referrer)?;

```

Comment:
Pretty sure resolve_import just affixes the base to the url; this can likely just always be done here, instead of in the trait

Reply:
That function is actually more precise than it looks. It's meant to implement this algorithm specifically: https://html.spec.whatwg.org/multipage/webappapis.html#resolve-a-module-specifier
What you're saying would probably work in most cases, but I personally think it'd be safer to just assume there's more edge cases covered by doing it this way.

Reply:
Yes I read the spec
That algorithm describes resolving a url with a base URL

And it should be done in all cases, it's why it's the first thing resolve does. It's required to be the first thing in resolve, therefore it needs to be outside the trait

Reply:
I think I misunderstood again. That function is also called outside of the trait. The one inside of the trait won't be needed once I change the resolve method of ImportProvider to return an Option.

Reply:
Yes but it needs to be called before passing the resulting URL to the trait to ensure the code properly accepts configured base urls

Reply:
Did you see my other comment about resolving relative paths to schemes other than file or http(s)? Is that just functionally not within scope for ImportProvider's resolve method?

Reply:
Yeah I completely misunderstood what that function does. I'll fix up this PR when I get back online later.